core/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointTrackerTests.java

```diff
@@ -540,17 +542,183 @@ public void testRaceUpdatingGlobalCheckpoint() throws InterruptedException, Brok
         assertThat(tracker.getGlobalCheckpoint(), equalTo((long) nextActiveLocalCheckpoint));
     }
 
+    public void testPrimaryContextHandoff() throws IOException {
+        GlobalCheckpointTracker oldPrimary = new GlobalCheckpointTracker(new ShardId("test", "_na_", 0),
+            IndexSettingsModule.newIndexSettings("test", Settings.EMPTY), UNASSIGNED_SEQ_NO);
+        GlobalCheckpointTracker newPrimary = new GlobalCheckpointTracker(new ShardId("test", "_na_", 0),
+            IndexSettingsModule.newIndexSettings("test", Settings.EMPTY), UNASSIGNED_SEQ_NO);
+
+        FakeClusterState clusterState = initialState();
+        clusterState.apply(oldPrimary);
+        clusterState.apply(newPrimary);
+
+        activatePrimary(clusterState, oldPrimary);
+
+        for (int i = 0; i < randomInt(10); i++) {
```

Comment:
nit: this samples the randomInt again and again.. much less likely to get to 10.